### PR TITLE
Add proxy app and counter contract

### PR DESCRIPTION
**Pull request by netoneko**
- [x] remove FIXME entries
- [x] add staging configuration for the contract
- [x] add app to build process and docker image

cc @liorkupf

### CODE REVIEW

File: projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/event-counter-contract.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import BaseSmartContract from "../base-smart-contact";
+
+export default class EventCounterContract extends BaseSmartContract {
+  public async reportEvent(event: string) {
+    const value: number = await this.getValue(event);
```
Comment (jlevison) [inline projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/event-counter-contract.ts]: you should verify that the 'event' is a string (in getCounter as well), see the kinatn (assuming thats the correct way of doing it)

File: projects/libs/core-library-typescript/test/virtual-machine/event-counter-smart-contract.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+let bobContract: BaseSmartContract;
+
+describe("event counter contract ", () => {
+  beforeEach(() => {
+    adapter = new ContractStateMemCacheAccessor(CONTRACT_ADDRESS, new StateCache());
+    aliceContract = new EventCounterContract(ALICE_ADDRESS, adapter);
+    bobContract = new EventCounterContract(BOB_ADDRESS, adapter);
+  });
+
+
```
Comment (jlevison) [inline projects/libs/core-library-typescript/test/virtual-machine/event-counter-smart-contract.spec.ts]: add a test for when you do not send a string and make sure it does not update the store. (2 as any) will override typescript checks and allow an int afaik

File: client/examples/event-proxy-app/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  return getRedis().hmsetAsync(username, { publicKey: keyPair.publicKey, privateKey: keyPair.getPrivateKeyUnsafe() }).timeout(config.timeout);
+}
+
+async function loadAccount(address: any) {
+  return getRedis().hgetallAsync(address).timeout(config.timeout);
+}
+
+const app = express();
+
+app.use("/", async (req: express.Request, res: express.Response) => {
```
Comment (jlevison) [inline client/examples/event-proxy-app/src/index.ts]: it should be a post, no? it looks like its implemented as a get
also, just making sure, there is no way to 'getCounter' via the proxy, right? (or did i miss it?)

File: client/examples/event-proxy-app/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+const VIRTUAL_CHAIN_ID = "6c696e";
+
+const {
+  ORBS_API_ENDPOINT,
+  TRANSACTION_TIMEOUT,
+  NETWORK_ID,
+  REDIS_URL
+} = process.env;
+
+const config = {
```
Comment (jlevison) [inline client/examples/event-proxy-app/src/index.ts]: maybe lock it down to the interface

File: deploy/bootstrap/.env [hunk 5]
```diff
@@ -4,4 +4,4 @@ GOSSIP_PEERS=ws://us-east-1.global.nodes.staging.orbs-test.com:60001,ws://eu-cen
 NUM_OF_NODES=6
 # Should NOT be enabled in production
 # CHANGE NETWORK ID FOR PRODUCTION TO "M"
-SMART_CONTRACTS_TO_LOAD=[{"vchainId": "640ed3", "name": "foobar", "filename": "foobar-smart-contract", "networkId": "T"},{"vchainId": "640ed3", "name": "text-message", "filename": "text-message-smart-contract", "networkId": "T"},{"vchainId": "6b696e", "name": "kinatn", "filename": "kinatn-smart-contract", "networkId": "T"}]
+SMART_CONTRACTS_TO_LOAD=[{"vchainId": "640ed3", "name": "foobar", "filename": "foobar-smart-contract", "networkId": "T"},{"vchainId": "640ed3", "name": "text-message", "filename": "text-message-smart-contract", "networkId": "T"},{"vchainId": "6b696e", "name": "kinatn", "filename": "kinatn-smart-contract", "networkId": "T"},{"vchainId": "6c696e", "name": "event-counter", "filename": "event-counter-smart-contract", "networkId": "T"}]
```
Comment (liorkupf) [inline deploy/bootstrap/.env]: isn't it event-counter-contract?

File: client/examples/event-proxy-app/src/index.ts (hunk 3 above)
Comment (netoneko) [reply client/examples/event-proxy-app/src/index.ts]: It's a get according to the spec, but I will allow it to be any event with these query params.

File: deploy/bootstrap/.env (hunk 5 above)
Comment (netoneko) [reply deploy/bootstrap/.env]: You are right, I will rename the file.

File: projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/event-counter-contract.ts (hunk 1 above)
Comment (netoneko) [reply projects/libs/core-library-typescript/src/virtual-machine/hard-coded-contracts/registry/event-counter-contract.ts]: Done.

File: projects/libs/core-library-typescript/test/virtual-machine/event-counter-smart-contract.spec.ts (hunk 2 above)
Comment (netoneko) [reply projects/libs/core-library-typescript/test/virtual-machine/event-counter-smart-contract.spec.ts]: Done.

**Pull request merged**